XiEditor/AppDelegate.swift

```diff
@@ -121,6 +121,18 @@ class AppDelegate: NSObject, NSApplicationDelegate {
                 guard let doc = doc as? Document else { continue }
                 doc.editViewController?.themeChanged(name)
             }
+        case "update_cmds":
+            guard let obj = params as? [String : AnyObject],
+                let view_id = obj["view_id"] as? String,
+                let document = documentForViewIdentifier(viewIdentifier: view_id),
+                let cmds = obj["cmds"] as? [[String: AnyObject]],
+                let plugin = obj["plugin"] as? String else { print("missing plugin field in \(params)"); return nil }
+            let parsedCommands = cmds.map { Command(jsonObject: $0) }
+                .filter { $0 != nil }
+                .map { $0! }
```

Comment:
Ah cool, that would feel much nicer. And yes, It was pretty frustrating writing all the json wrangling stuff knowing I'm going to tear it all out in a month or two. 😒